source/frontend/src/components/maps/leaflet/Layers/PointClusterer.tsx

```diff
@@ -252,122 +259,135 @@ export const PointClusterer: React.FC<React.PropsWithChildren<PointClustererProp
     }
   }, [featureGroupRef, mapInstance, clusters, selectedFeature, tilesLoaded]);
 
-  return (
-    <>
-      <FeatureGroup ref={featureGroupRef}>
-        {/**
-         * Render all visible clusters
-         */}
-        {clusters.map((cluster, index) => {
-          // every cluster point has coordinates
-          const [longitude, latitude] = cluster.geometry.coordinates;
-
-          // Only clusters have the cluster property, if so we have a cluster to render
-          if ('cluster' in cluster.properties) {
-            const clusterFeature = cluster as ClusterFeature<ClusterProperties>;
-            const { point_count: pointCount, point_count_abbreviated } = clusterFeature.properties;
-
-            const sizeClass = pointCount < 100 ? 'small' : pointCount < 1000 ? 'medium' : 'large';
-            return (
-              // render the cluster marker
-              <Marker
-                key={index}
-                position={[latitude, longitude]}
-                eventHandlers={{
-                  click: e => {
-                    zoomOrSpiderfy(clusterFeature);
-                    e.target.closePopup();
-                  },
-                }}
-                icon={
-                  new L.DivIcon({
-                    html: `<div><span>${point_count_abbreviated}</span></div>`,
-                    className: `marker-cluster marker-cluster-${sizeClass}`,
-                    iconSize: [40, 40],
-                  })
-                }
-              />
-            );
-          } else {
-            const clusterFeature = cluster as PointFeature<
+  const mapMarkerClickFn = mapMachine.mapMarkerClick;
+  const renderedPoints = useMemo(() => {
```

Comment:
In normal situations, react handles the prevention of unnecessary renders to the render tree. However, it may not be able to do so here due to interactions with leaflet - as a result, this render method was greatly slowing down the overall map render time.

Reply:
(without this memo, this render method is called 4-6 times per user action)